### PR TITLE
X1084 add costing to xenium probe hybe

### PR DESCRIPTION
**Pull request by sabrine33**

### CODE REVIEW

File: src/components/probeHybridisation/ProbeAddPanel.tsx [hunk 1]
```diff
@@ -145,7 +186,15 @@ const ProbeAddPanel = ({ probePanels }: ProbeLotAddPanelProps) => {
...
                   const updatedLabware: ProbeOperationLabware = {
                     ...lw,
-                    probes: [...lw.probes, { name: probeName, lot: probeLot, plex: Number(probePlex) }]
+                    probes: [
+                      ...lw.probes,
+                      {
+                        name: probeName,
+                        lot: probeLot,
+                        plex: Number(probePlex),
+                        costing: probeCosting === 'SGP' ? SlideCosting.Sgp : SlideCosting.Faculty
```
Comment (seenanair) [inline src/components/probeHybridisation/ProbeAddPanel.tsx:195]: As probeCosting can be empty as well, it is better to compare with correct value other than ternary operation.

File: src/lib/machines/probeHybridisation/probeHybridisationMachine.ts [hunk 2]
```diff
@@ -181,7 +182,7 @@ export const probeHybridisationMachine = createMachine<ProbeHybridisationContext
           if (!ctx.probeLabware.some((plw) => plw.barcode === lw.barcode)) {
             ctx.probeLabware.push({
               barcode: lw.barcode,
-              probes: [{ name: '', lot: '', plex: -1 }],
+              probes: [{ name: '', lot: '', plex: -1, costing: SlideCosting.Sgp }],
```
Comment (seenanair) [inline src/lib/machines/probeHybridisation/probeHybridisationMachine.ts:185]: I think costing has to be an empty string as initial value

File: src/lib/machines/probeHybridisation/probeHybridisationMachine.ts [hunk 3]
```diff
@@ -215,7 +216,7 @@ export const probeHybridisationMachine = createMachine<ProbeHybridisationContext
         if (e.type !== 'ADD_PROBE_LOT') return;
         const probeLabware = ctx.probeLabware.find((plw) => plw.barcode === e.barcode);
         if (!probeLabware) return;
-        probeLabware.probes.push({ name: '', lot: '', plex: -1 });
+        probeLabware.probes.push({ name: '', lot: '', plex: -1, costing: SlideCosting.Sgp });
```
Comment (seenanair) [inline src/lib/machines/probeHybridisation/probeHybridisationMachine.ts:219]: Same here as well, costing field should be empty initially

File: src/components/probeHybridisation/ProbeAddPanel.tsx (hunk 1 above)
Comment (sabrine33) [reply src/components/probeHybridisation/ProbeAddPanel.tsx:195]: probeCosting is a required field can not be empty, validation will not pass until the user selects a costing

Comment (seenanair) [reply src/components/probeHybridisation/ProbeAddPanel.tsx:195]: Ok

File: src/lib/machines/probeHybridisation/probeHybridisationMachine.ts (hunk 3 above)
Comment (sabrine33) [reply src/lib/machines/probeHybridisation/probeHybridisationMachine.ts:219]: This is due to the type restriction,  I can not set the costing the empty/null as it is of tyoe SlideCosting, on the ui when the user adds a new probe row, the select box is set to empty

File: src/lib/machines/probeHybridisation/probeHybridisationMachine.ts (hunk 2 above)
Comment (sabrine33) [reply src/lib/machines/probeHybridisation/probeHybridisationMachine.ts:185]: This is due to the type restriction,  I can not set the costing the empty/null as it is of tyoe SlideCosting, on the ui when the user adds a new probe row, the select box is set to empty

File: src/lib/machines/probeHybridisation/probeHybridisationMachine.ts (hunk 3 above)
Comment (seenanair) [reply src/lib/machines/probeHybridisation/probeHybridisationMachine.ts:219]: Why can't the type be defined as SlideCosting | string or SlideCosting|undefined?

**Review by seenanair — APPROVED** [review]

**Pull request merged**